setup.go

```diff
@@ -134,14 +134,22 @@ func Setup(ctx context.Context, cfg Config, key crypto.PrivKey, dnsCache *cached
 	}
 
 	opts := []libp2p.Option{
-		libp2p.ListenAddrStrings(cfg.ListenAddrs...),
 		libp2p.NATPortMap(),
 		libp2p.ConnectionManager(cmgr),
 		libp2p.Identity(key),
 		libp2p.BandwidthReporter(bwc),
 		libp2p.DefaultTransports,
 		libp2p.DefaultMuxers,
 		libp2p.ResourceManager(bitswapRcMgr),
+		libp2p.EnableHolePunching(),
+	}
+
+	if len(cfg.ListenAddrs) == 0 {
```

Comment:
This is **always** true. We set `cfg.ListenAddrs` and `cfg.AnnounceAddrs` nowhere in the code. Hence me removing them in #111. I would just replace this with:

```go
opts := []libp2p.Option{
	libp2p.DefaultListenAddrs,
	//...
```

And remove the code below for the announce addresses.